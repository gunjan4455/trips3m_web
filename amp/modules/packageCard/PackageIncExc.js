/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';

import { capitalizeFirstLetter } from 'amp/utils/parsers';
import { InclusionNotification } from 'constants/Packages/InclusionNotification';
import YellowDot from 'amp/modules/common/YellowDot';
import 'amp/screens/pdp/priceModule/PackageIncExc.scss';
import IconsMapping from 'amp/helpers/Icon/IconMappings';

const getIconText = (inclusion, hotelInfo) => {
  if (inclusion.key === 'hotel' && hotelInfo) {
    return `Upto ${hotelInfo[0]} Stars`;
  }
  return inclusion.text;
};

const PackageIncExc = ({ inclusions, id, noScroll, hotelInfo, isPackageRevamp }) => {
  const inclusionsAval = isPackageRevamp ? inclusions.filter(inclusion =>
    inclusion.available).slice(0,4) : inclusions;

  const inclusionList = inclusionsAval.map(
    (inclusion, index) => {
      const iconName = IconsMapping[capitalizeFirstLetter(inclusion.key.replace(/[\s-_]/g, ''))] || 'DefaultIcon';
      const disabledElementClass = inclusion.available ? '' : 'not-included';
      const disabledIconClass = inclusion.available ? '' : 'greyscale';
      const inclusionTextHidden = inclusion.text ? '' : 'hide';
      const InclusionNotificationPresent = (inclusion.available && (InclusionNotification.indexOf(inclusion.key) > -1)) ? 'relative' : '';

      return (
        <div className={`incl-excl-box ${inclusionTextHidden} ${disabledElementClass} ${InclusionNotificationPresent}`} key={`${id}_${inclusion.key}_${index}`}>
          {
            (inclusion.available && (InclusionNotification.indexOf(inclusion.key) > -1)) ?
            <YellowDot id="flightMessage" dotBoxStyle={{ width: '10px', height: '10px', position: 'absolute', top: '6px', right: '3px' }} dotSmallBoxStyle={{ width: '5px', height: '5px' }} spinnerStyle={{ width: '10px', height: '10px', zIndex: '1' }}  hoverData="Flight prices are dynamic and may change at the time of booking" />
            : null
          }
          <span className="incl-excl-img" style={{background: `url('https://js.traveltriangle.com/public-product/amp-icons-v0.svg#${iconName}-usage')`}}>
          </span>
          <p className="f10">{getIconText(inclusion, hotelInfo)}</p>
        </div>
      );
    }
  );

  return (
    <div
      className={`row row- incl-excl-con ${noScroll ? 'inc-no-scroll' : ''} `}
    >
      <div className="row row- incl-excl-con-inner">
        {inclusionList}
      </div>
    </div>
  );
};

PackageIncExc.propTypes = {
  inclusions: PropTypes.array.isRequired,
  hotelInfo: PropTypes.array.isRequired,
  id: PropTypes.string,
  noScroll: PropTypes.bool,
  isPackageRevamp: PropTypes.bool
};

PackageIncExc.defaultProps = {
  noScroll: false,
  id: 'default_1',
  isPackageRevamp: false
};


export default PackageIncExc;
