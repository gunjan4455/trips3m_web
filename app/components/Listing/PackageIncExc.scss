.incl-excl-con {
    overflow-x: scroll;
    width: 100%;
    padding-bottom: 30px;
}

.incl-excl-con-inner {
    margin: 0;
    padding: 0;
    display: -webkit-box;
    display: flex;
}

.package-card {
    .incl-excl-con-inner.incl-excl-con-inner-deals {
        height: auto;
    }
}

.incl-excl-box {
    max-width: 22%;
    min-width: 22%;
    padding: 0;
    word-wrap: break-word;
    display: table-cell;
    text-align: center;
}

.incl-excl-box.not-included {
    opacity: 0.4;
}

.incl-excl-box.not-included .incl-excl-img::before {
    content: '';
    box-sizing: content-box;
    width: 2px;
    height: 26px;
    border-left: 2px solid #FFFFFF;
    border-right: 2px solid #FFFFFF;
    background: #B2B2B2;
    position: absolute;
    display: block;
    bottom: -6px;
    right: 11px;
    transform: rotate(-25deg);
    z-index: 4;
}

.incl-excl-img {
    width: 30px;
    height: 30px;
    display: block;
    margin: 0 auto;
    margin-bottom: 8px;
    background: #ffffff;
    position: relative;
}

.incl-excl-img svg {
    left: 50%;
    -moz-transform: translateX(-50%);
    -webkit-transform: translateX(-50%);
    transform: translateX(-50%);
    width: auto;
    max-width: 100%;
    max-height: 20px;
    position: absolute;
    bottom: 0;
}

.incl-excl-con-inner p {
    line-height: 12px;
}

.inc-no-scroll {
    .incl-excl-box {
        max-width: 20%;
        min-width: 20%;
        height: 62px;
        overflow: hidden;
    }
}

.inclusionDeals {
    height: 24px;
}