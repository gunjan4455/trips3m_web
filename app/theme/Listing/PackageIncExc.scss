.package-incexc-list {
  margin: 0;
  overflow: hidden;
  padding: 0;
  li  {
    list-style: none;
    display: inline-block;
    width: 55px;
    margin: 0 5px;
    float: left;
    text-align: center;
    height: auto;
    max-height: 66px;
    overflow: hidden;
  }
}

.package-incexc-list li.not-included  {
  opacity: 0.4;
}
.package-incexc-list li.not-included .icon-box svg {
  -webkit-filter: grayscale(100%);
  filter: grayscale(100%);
}
.package-incexc-list li.not-included .icon-box::before {
  content: '';
  box-sizing: content-box;
  width: 2px;
  height: 26px;
  border-left: 2px solid #FFFFFF;
  border-right: 2px solid #FFFFFF;
  background: #B2B2B2;
  position: absolute;
  display: block;
  bottom: -3px;
  right: 11px;
  transform: rotate(-25deg);
  z-index: 4;
}
.package-incexc-list li .icon-box {
  width: 24px;
  display: block;
  margin:0 auto;
  height: 24px;
  position: relative;
}
.package-incexc-list li .icon-box svg {
  left:50%;
  -moz-transform: translateX(-50%);
  -webkit-transform: translateX(-50%);
  transform: translateX(-50%);
  width: auto;
  max-width: 100%;
  height: 20px;
  position: absolute;
  bottom: 0;
  display: block;
  top: auto;
}
.package-incexc-list p {line-height: 12px; font-size: 10px; padding-top: 5px;
word-wrap: break-word;color:#3E3E3E;}
