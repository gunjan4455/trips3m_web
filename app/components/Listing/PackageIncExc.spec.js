import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import PackageIncExc from './PackageIncExc';

configure({ adapter: new Adapter() });

const props = {
  'inclusions': [{ 'key': 'hotel', 'available': true, 'text': 'Upto 3 Stars' }, {
    'key': 'flights',
    'available': false,
    'text': 'Flights'
  }, { 'key': 'meals', 'available': true, 'text': 'Meals' }, {
    'key': 'sightseeing',
    'available': true,
    'text': 'Sightseeing'
  }, { 'key': 'transfer', 'available': true, 'text': 'Airport Transfer' }, {
    'key': 'entrytickets',
    'available': true,
    'text': 'Entry Tickets'
  }, { 'key': 'visa', 'available': true, 'text': 'Visa' }],
  'id': 'card_334',
  'hotelInfo': [5, 49000, 53900],
  'noScroll': false
};

const wrapper = shallow(<PackageIncExc {...props} />);

describe('PackageIncExc test suit', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});


