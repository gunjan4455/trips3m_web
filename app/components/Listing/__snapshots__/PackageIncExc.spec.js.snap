// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackageIncExc test suit should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PackageIncExc
    hotelInfo={
      Array [
        5,
        49000,
        53900,
      ]
    }
    id="card_334"
    inclusions={
      Array [
        Object {
          "available": true,
          "key": "hotel",
          "text": "Upto 3 Stars",
        },
        Object {
          "available": false,
          "key": "flights",
          "text": "Flights",
        },
        Object {
          "available": true,
          "key": "meals",
          "text": "Meals",
        },
        Object {
          "available": true,
          "key": "sightseeing",
          "text": "Sightseeing",
        },
        Object {
          "available": true,
          "key": "transfer",
          "text": "Airport Transfer",
        },
        Object {
          "available": true,
          "key": "entrytickets",
          "text": "Entry Tickets",
        },
        Object {
          "available": true,
          "key": "visa",
          "text": "Visa",
        },
      ]
    }
    isPackageRevamp={false}
    noScroll={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row row- incl-excl-con-inner"
      >
        <div
          className="incl-excl-box   "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className=""
            />
          </span>
          <p
            className="f10"
          >
            Upto 5 Stars
          </p>
        </div>
        <div
          className="incl-excl-box  not-included "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className="greyscale"
            />
          </span>
          <p
            className="f10"
          >
            Flights
          </p>
        </div>
        <div
          className="incl-excl-box   "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className=""
            />
          </span>
          <p
            className="f10"
          >
            Meals
          </p>
        </div>
        <div
          className="incl-excl-box   "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className=""
            />
          </span>
          <p
            className="f10"
          >
            Sightseeing
          </p>
        </div>
        <div
          className="incl-excl-box   "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className=""
            />
          </span>
          <p
            className="f10"
          >
            Airport Transfer
          </p>
        </div>
        <div
          className="incl-excl-box   "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className=""
            />
          </span>
          <p
            className="f10"
          >
            Entry Tickets
          </p>
        </div>
        <div
          className="incl-excl-box   "
        >
          <span
            className="incl-excl-img"
          >
            <IconSvg
              className=""
            />
          </span>
          <p
            className="f10"
          >
            Visa
          </p>
        </div>
      </div>,
      "className": "row row- incl-excl-con  ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Upto 5 Stars
            </p>
          </div>,
          <div
            className="incl-excl-box  not-included "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className="greyscale"
              />
            </span>
            <p
              className="f10"
            >
              Flights
            </p>
          </div>,
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Meals
            </p>
          </div>,
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Sightseeing
            </p>
          </div>,
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Airport Transfer
            </p>
          </div>,
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Entry Tickets
            </p>
          </div>,
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Visa
            </p>
          </div>,
        ],
        "className": "row row- incl-excl-con-inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "card_334_hotel_0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>,
              <p
                className="f10"
              >
                Upto 5 Stars
              </p>,
            ],
            "className": "incl-excl-box   ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className=""
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_hotel_0",
                "nodeType": "function",
                "props": Object {
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Upto 5 Stars",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Upto 5 Stars",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "card_334_flights_1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className="greyscale"
                />
              </span>,
              <p
                className="f10"
              >
                Flights
              </p>,
            ],
            "className": "incl-excl-box  not-included ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className="greyscale"
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_flights_1",
                "nodeType": "function",
                "props": Object {
                  "className": "greyscale",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Flights",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Flights",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "card_334_meals_2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>,
              <p
                className="f10"
              >
                Meals
              </p>,
            ],
            "className": "incl-excl-box   ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className=""
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_meals_2",
                "nodeType": "function",
                "props": Object {
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Meals",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Meals",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "card_334_sightseeing_3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>,
              <p
                className="f10"
              >
                Sightseeing
              </p>,
            ],
            "className": "incl-excl-box   ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className=""
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_sightseeing_3",
                "nodeType": "function",
                "props": Object {
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sightseeing",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Sightseeing",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "card_334_transfer_4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>,
              <p
                className="f10"
              >
                Airport Transfer
              </p>,
            ],
            "className": "incl-excl-box   ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className=""
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_transfer_4",
                "nodeType": "function",
                "props": Object {
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Airport Transfer",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Airport Transfer",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "card_334_entrytickets_5",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>,
              <p
                className="f10"
              >
                Entry Tickets
              </p>,
            ],
            "className": "incl-excl-box   ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className=""
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_entrytickets_5",
                "nodeType": "function",
                "props": Object {
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Entry Tickets",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Entry Tickets",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "card_334_visa_6",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>,
              <p
                className="f10"
              >
                Visa
              </p>,
            ],
            "className": "incl-excl-box   ",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <IconSvg
                  className=""
                />,
                "className": "incl-excl-img",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "card_334_visa_6",
                "nodeType": "function",
                "props": Object {
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Visa",
                "className": "f10",
              },
              "ref": null,
              "rendered": "Visa",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row row- incl-excl-con-inner"
        >
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Upto 5 Stars
            </p>
          </div>
          <div
            className="incl-excl-box  not-included "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className="greyscale"
              />
            </span>
            <p
              className="f10"
            >
              Flights
            </p>
          </div>
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Meals
            </p>
          </div>
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Sightseeing
            </p>
          </div>
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Airport Transfer
            </p>
          </div>
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Entry Tickets
            </p>
          </div>
          <div
            className="incl-excl-box   "
          >
            <span
              className="incl-excl-img"
            >
              <IconSvg
                className=""
              />
            </span>
            <p
              className="f10"
            >
              Visa
            </p>
          </div>
        </div>,
        "className": "row row- incl-excl-con  ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="incl-excl-box   "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>
              <p
                className="f10"
              >
                Upto 5 Stars
              </p>
            </div>,
            <div
              className="incl-excl-box  not-included "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className="greyscale"
                />
              </span>
              <p
                className="f10"
              >
                Flights
              </p>
            </div>,
            <div
              className="incl-excl-box   "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>
              <p
                className="f10"
              >
                Meals
              </p>
            </div>,
            <div
              className="incl-excl-box   "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>
              <p
                className="f10"
              >
                Sightseeing
              </p>
            </div>,
            <div
              className="incl-excl-box   "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>
              <p
                className="f10"
              >
                Airport Transfer
              </p>
            </div>,
            <div
              className="incl-excl-box   "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>
              <p
                className="f10"
              >
                Entry Tickets
              </p>
            </div>,
            <div
              className="incl-excl-box   "
            >
              <span
                className="incl-excl-img"
              >
                <IconSvg
                  className=""
                />
              </span>
              <p
                className="f10"
              >
                Visa
              </p>
            </div>,
          ],
          "className": "row row- incl-excl-con-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "card_334_hotel_0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className=""
                  />
                </span>,
                <p
                  className="f10"
                >
                  Upto 5 Stars
                </p>,
              ],
              "className": "incl-excl-box   ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className=""
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_hotel_0",
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Upto 5 Stars",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Upto 5 Stars",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "card_334_flights_1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className="greyscale"
                  />
                </span>,
                <p
                  className="f10"
                >
                  Flights
                </p>,
              ],
              "className": "incl-excl-box  not-included ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className="greyscale"
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_flights_1",
                  "nodeType": "function",
                  "props": Object {
                    "className": "greyscale",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Flights",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Flights",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "card_334_meals_2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className=""
                  />
                </span>,
                <p
                  className="f10"
                >
                  Meals
                </p>,
              ],
              "className": "incl-excl-box   ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className=""
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_meals_2",
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Meals",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Meals",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "card_334_sightseeing_3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className=""
                  />
                </span>,
                <p
                  className="f10"
                >
                  Sightseeing
                </p>,
              ],
              "className": "incl-excl-box   ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className=""
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_sightseeing_3",
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sightseeing",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Sightseeing",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "card_334_transfer_4",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className=""
                  />
                </span>,
                <p
                  className="f10"
                >
                  Airport Transfer
                </p>,
              ],
              "className": "incl-excl-box   ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className=""
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_transfer_4",
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Airport Transfer",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Airport Transfer",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "card_334_entrytickets_5",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className=""
                  />
                </span>,
                <p
                  className="f10"
                >
                  Entry Tickets
                </p>,
              ],
              "className": "incl-excl-box   ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className=""
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_entrytickets_5",
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Entry Tickets",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Entry Tickets",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "card_334_visa_6",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                null,
                <span
                  className="incl-excl-img"
                >
                  <IconSvg
                    className=""
                  />
                </span>,
                <p
                  className="f10"
                >
                  Visa
                </p>,
              ],
              "className": "incl-excl-box   ",
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <IconSvg
                    className=""
                  />,
                  "className": "incl-excl-img",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "card_334_visa_6",
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Visa",
                  "className": "f10",
                },
                "ref": null,
                "rendered": "Visa",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
